test/src/e2e_vm_tests/test_programs/should_pass/supertraits_for_abis_ownable/src/main.sw

```diff
@@ -0,0 +1,61 @@
+contract;
```

Comment:
I feel like we can make this example more realistic (and also use it in the `examples` folder) as follows. In a library file, create a helper trait for accessing a storage variable called `ownable`:
```rust
trait StorageHelpers {
    fn get_owner() -> b256;
    
    fn set_owner(owner: b256);
}
```
Then, create an abi called `Ownable` as follows:
```rust
abi Ownable {
    // Nothing here
} {
    #[storage(read)]
    fn owner() -> b256 {
        Self::get_owner() // Or just `get_owner()` if that works
    }

    // and so on - key is not to use `storage.` here because we're meant to be in a library
}
```
Now, in the main contract, we can write the following:
```rust
storage {
    owner: b256 = ZERO_B256, // Actual owner
}

impl StorageHelpers for Contract {
    fn get_owner() -> b256 {
        storage.owner
    }
    
    fn set_owner(owner: b256) {
        storage.owner = owner;
    }
}

impl Ownable for Contract { }
````
Now that I'm writing this, I see that we're missing a feature to make this fully usable. That feature is **allowing `abi`s to be "superabis" of other `abi`s**. If we enable that feature, we can then write the following:
```rust
// Implement `StorageHelpers` for `Contract

// Main contract `abi`:
abi MyContract : Ownable {
    fn foo();
}

// Implement the contract:
impl Ownable for Contract { }

impl MyContract for Contract {
    fn foo() {
        // implement `foo`
    }
}
```
Now, if we were to use our contract in a script (for that, we have to move the definition of `MyContract` to a separate library that the script can import), we would be able to do:
```rust
fn main() {
    let caller = abi(MyContract, <contract_id>);

    let owner = caller.owner(); // We're able to call `owner` from `abi Ownable`
    caller.foo(); // We're also able to call `foo`
}
```
The same applies to the SDKs. That is, we would be able to call methods from `Ownable` and `MyContract` from the SDK by simply loading the JSON ABI` of `MyContract`.

Reply:
@mohammadfawaz Great suggestions! I just pushed some changes, are those close to what you have in mind?

Reply:
Thanks @anton-trunov. Almost, but not quite :) Basically, the idea is to have `Ownable` be an `abi` and not a `trait`, so that its methods are accessible externally. And the main contract can then be `abi MyContrat: Ownable` which implies that we need abi as superabis of other abis which we don't have yet. I think what you have for now is reasonable, though it doesn't fully solve the "storage in libraries" problem yet, at least not in its full form.  I suggest we look into what it takes to support abis as "superabis".

Reply:
Got it. Opened an issue to track this suggestion: https://github.com/FuelLabs/sway/issues/4049